public/app/features/canvas/runtime/sceneAbleManagement.ts

```diff
@@ -33,6 +33,38 @@ const enableCustomables = (moveable: Moveable) => {
   };
 };
 
+/* 
+  Helper function that determines if the selected DOM target is currently selected in selecto state.
+
+  For context canvas elements each have a different level of nesting.
+  Given this, we need to traverse up the DOM tree from the selected target to find
+  the element's registered selecto div to determine if the selected target is already selected in selecto state.
+  See `initMoveable` and `generateTargetElements` for more context.
+*/
+const isTargetAlreadySelected = (selectedTarget: HTMLElement, scene: Scene) => {
+  let selectedTargetParent = selectedTarget.parentElement;
+  let isTargetAlreadySelected = false;
+
+  // Traverse up the DOM tree to check if the selected target is already selected
+  while (selectedTargetParent) {
```

Comment:
This makes me nervous. I know it should break when we hit scene root, and beyond that at Document, but perhaps there is a more direct way to do this loop that involves a depth counter.

Reply:
where would we have this depth counter? 🤔 I'm not sure of a better way of handling this given inconsistent nesting / DOM structure of canvas elements

Reply:
I thought maybe we already had one during grouping events.😳

I think your approach is okay, but can we test that it does in fact exit the while when reaching Document by removing the scene root break?

Reply:
in my testing it never reached edge case of going up to scene.root.div (mainly due to the current target is coming from a selecto drag start event which is are limited to canvas elements 🤔) These are initialized in the `initMovable` function via https://github.com/grafana/grafana/blob/1534329fe9a038c07cf3ead2ecdbe1049a882e84/public/app/features/canvas/runtime/sceneAbleManagement.ts#L71